### PR TITLE
Upgrade to Oracle Database 23c Free

### PR DESCRIPTION
**Pull request by gvenzl**
### Description
This PR updates the unittest ReadMe and `setup.cfg` to use Oracle Database Free, the successor to Oracle Database XE.

### Checklist

This pull request is:

- [ ] A documentation / typographical / small typing error fix
	- Good to go, no issue or tests are needed
- [ ] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -179,7 +179,7 @@ mssql = mssql+pyodbc://scott:tiger^5HHH@mssql2017:1433/test?driver=ODBC+Driver+1
...
 pymssql = mssql+pymssql://scott:tiger^5HHH@mssql2017:1433/test
 docker_mssql = mssql+pyodbc://scott:tiger^5HHH@127.0.0.1:1433/test?driver=ODBC+Driver+18+for+SQL+Server
-oracle = oracle+cx_oracle://scott:tiger@oracle18c/xe
-cxoracle = oracle+cx_oracle://scott:tiger@oracle18c/xe
-oracledb = oracle+oracledb://scott:tiger@oracle18c/xe
-docker_oracle = oracle+cx_oracle://scott:tiger@127.0.0.1:1521/?service_name=XEPDB1
+oracle = oracle+cx_oracle://scott:tiger@oracle23c/free
+cxoracle = oracle+cx_oracle://scott:tiger@oracle23c/free
+oracledb = oracle+oracledb://scott:tiger@oracle23c/free
+docker_oracle = oracle+cx_oracle://scott:tiger@127.0.0.1:1521/?service_name=FREEPDB1
```
Comment (CaselIT) [inline setup.cfg:185]: other than the docker one, the other are used by the ci, so let's revert that bit

Comment (CaselIT) [reply setup.cfg:185]: (at least I think, cc @zzzeek )

Comment (zzzeek) [reply setup.cfg:185]: the setup.cfg entries are biased towards....me personally!   oracle18c is the hostname I have here in our CI environment.   so we can update the README but until I update CI can we please revert the setup.cfg part, thanks!

**Review by zzzeek — CHANGES_REQUESTED** [review]

File: setup.cfg (hunk 1 above)
Comment (CaselIT) [reply setup.cfg:185]: IIRC docker_oracle is fine to update, since it's just something I added when adding the docker example that's been updated here

**Comment by gvenzl** [comment]
Hi both, thanks a lot for considering this PR.

Hm, if `oracle18c` is the name of the CI server, should I leave them everywhere `oracle18c` including in the ReadMe?

**Comment by zzzeek** [comment]
I think what should happen is, I update our CI server to oracle 23, then we dont have to worry about it

for now, in the README where it shows the current output of "--dbs", sure keep it as 18c since that's what it will say at the moment, if that's waht you mean

File: setup.cfg (hunk 1 above)
Comment (gvenzl) [reply setup.cfg:185]: Reverted the CI server entries back to `oracle18c/xe`

**Comment by gvenzl** [comment]
Ok, I reverted the setup.cfg entries for the CI server back to `oracle18c/xe` and updated the ReadMe to refer `oracle18c` again.

**Comment by CaselIT** [comment]
thanks!

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 1cfb2d3a7f688a4b887182d2485b7d60f12eb013 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 1cfb2d3a7f688a4b887182d2485b7d60f12eb013: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4984

**Comment by CaselIT** [comment]
the CI seems happy

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4984 has been **merged**. Congratulations! :)

**Comment by gvenzl** [comment]
Thanks a lot! :)

**Comment by CaselIT** [comment]
thank you for the docker images :) It has made testing locally using oracle possible :)

**Pull request closed without merging**